extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/AvroFeatureDataAdapter.java

```diff
@@ -0,0 +1,123 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import mil.nga.giat.geowave.adapter.vector.plugin.visibility.JsonDefinitionColumnVisibilityManagement;
+import mil.nga.giat.geowave.adapter.vector.plugin.visibility.AvroAdaptorProxyFieldLevelVisibilityHandler;
+import mil.nga.giat.geowave.adapter.vector.utils.TimeDescriptors;
+import mil.nga.giat.geowave.core.geotime.store.dimension.Time;
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.store.adapter.IndexFieldHandler;
+import mil.nga.giat.geowave.core.store.adapter.NativeFieldHandler;
+import mil.nga.giat.geowave.core.store.adapter.NativeFieldHandler.RowBuilder;
+import mil.nga.giat.geowave.core.store.adapter.PersistentIndexFieldHandler;
+import mil.nga.giat.geowave.core.store.data.field.FieldReader;
+import mil.nga.giat.geowave.core.store.data.field.FieldVisibilityHandler;
+import mil.nga.giat.geowave.core.store.data.field.FieldWriter;
+import mil.nga.giat.geowave.core.store.data.visibility.VisibilityManagement;
+import mil.nga.giat.geowave.core.store.index.CommonIndexValue;
+
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+/**
+ */
+public class AvroFeatureDataAdapter extends
+		FeatureDataAdapter
+{
+
+	protected AvroFeatureDataAdapter() {}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>());
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final VisibilityManagement<SimpleFeature> visibilityManagement ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>(),
+				null,
+				visibilityManagement);
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final List<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>> customIndexHandlers ) {
+		super(
+				type,
+				customIndexHandlers,
+				null,
+				new JsonDefinitionColumnVisibilityManagement<SimpleFeature>());
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final FieldVisibilityHandler<SimpleFeature, Object> fieldVisiblityHandler ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>(),
+				fieldVisiblityHandler,
+				new JsonDefinitionColumnVisibilityManagement<SimpleFeature>());
+	}
+
+	@Override
+	protected List<NativeFieldHandler<SimpleFeature, Object>> typeToFieldHandlers(
+			final SimpleFeatureType type ) {
+		final List<NativeFieldHandler<SimpleFeature, Object>> nativeHandlers = new ArrayList<NativeFieldHandler<SimpleFeature, Object>>(
+				1);
+
+		nativeHandlers.add(new AvroFeatureAttributeHandler());
+		return nativeHandlers;
+	}
+
+	@Override
+	protected IndexFieldHandler<SimpleFeature, Time, Object> getTimeRangeHandler(
```

Comment:
makes sense, I created this before extending FeatureDataAdapter.  I will remove the override and the unnecessary avro handler class.